test/scripts/upgrade-ovn.sh

```diff
@@ -6,6 +6,17 @@ set -ex
 export KUBECONFIG=${HOME}/ovn.conf
 export OVN_IMAGE=${OVN_IMAGE:-ovn-daemonset-f:pr}
 
+detect_apiserver_url() {
+  # Detect API_URL used for in-cluster communication
+  #
+  # Despite OVN run in pod they will only obtain the VIRTUAL apiserver address
+  # and since OVN has to provide the connectivity to service
+  # it can not be bootstrapped
+  #
+  # This is the address of the node with the control-plane
+  API_URL=$(kind get kubeconfig --internal --name "${KIND_CLUSTER_NAME}" | grep server | awk '{ print $2 }')
```

Comment:
where is `API_URL` used ?

Reply:
So this is set right before rendering `ovn-setup.yaml.j2` after line 271:

```
# create upgraded OVN manifests from checked-out branch
set_cluster_cidr_ip_families
detect_apiserver_url
create_ovn_kube_manifests
install_ovn_image
```

Besides `net_cidr` and `svc_cidr`, there was another field (`k8s_apiserver`) in the `ovn-config` configmap that was filled in with a different value than what was set by `kind.sh`, so I repeated here the steps used in`kind.sh` to get the same value . 

In `create_ovn_kube_manifests`, `API_URL` is passed to `daemonset.sh`, which will render the yaml:
```
create_ovn_kube_manifests() {
  pushd ../dist/images
  ./daemonset.sh \
    --image="${OVN_IMAGE}" \
    --net-cidr="${NET_CIDR}" \
    --svc-cidr="${SVC_CIDR}" \
    --gateway-mode="${OVN_GATEWAY_MODE}" \
    --hybrid-enabled="${OVN_HYBRID_OVERLAY_ENABLE}" \
    --disable-snat-multiple-gws="${OVN_DISABLE_SNAT_MULTIPLE_GWS}" \
    --disable-pkt-mtu-check="${OVN_DISABLE_PKT_MTU_CHECK}" \
    --ovn-empty-lb-events="${OVN_EMPTY_LB_EVENTS}" \
    --multicast-enabled="${OVN_MULTICAST_ENABLE}" \
    --k8s-apiserver="${API_URL}" \
    ...
```

Reply:
@msherif1234 on a second thought, I removed this part altogether. It's only needed by `daemonset.sh` in order to render `ovn-setup.yaml.j2` and, when called from the upgrade test, I'm skipping on purpose the part of the yaml that includes the API server URL.

No need to complicate things. :)